### ISSUE TITLE
Request ID logging

### ISSUE THREAD

**Issue by der** [epimorphics/elda#232]
Log x-request-id header when present to allow log tracing from proxy.

Part of: https://github.com/epimorphics/hmlr-ansible-deployment/issues/93

**Comment by simonoakesepimorphics** [comment]
This is already implemented but needs to be enabled by setting the `ELDA_USE_ID` env var to "true" @der

**Comment by simonoakesepimorphics** [comment]
Can this be closed? @der

**Comment by der** [comment]
Leave it for the moment until it's deployed and checked, I'll close the ticket once that's done.

**Comment by der** [comment]
Tested in NRW (Elda 3.0.0) with no effect but neither NRW nor HMLR are set up to use json logging and presume that is required or some custom config of the formatter that isn't currently in either deployment

Is there an example Elaa log configuration which shows request_id being emited in some form that we start from?

**Comment by simonoakesepimorphics** [comment]
The request ID is set as a MDC field (request_id) so if the log format doesn't include the field then it's not visible. I mistakenly thought that the ID would be visible in the output because it was set in the MDC but the specific method doesn't support it for some reason - I've fixed this in the PR.

**Comment by simonoakesepimorphics** [comment]
The logback config I used to test this:
```
<configuration>
  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
    <encoder class="net.logstash.logback.encoder.LogstashEncoder">
      <timeZone>UTC</timeZone>
      <includeMdc>true</includeMdc>
    </encoder>
  </appender>
  <root level="info">
    <appender-ref ref="STDOUT" />
  </root>
</configuration>
```

**Comment by der** [comment]
This has not been tested yet.

**Comment by der** [comment]
Is there going to be a SNAPSHOT build to test or do I need to do all that locally?

**Comment by simonoakesepimorphics** [comment]
I've published a snapshot: s3://epi-repository/snapshot/com/epimorphics/lda/elda-lda/3.0.1-SNAPSHOT/elda-lda-3.0.1-20260618.152025-1.jar

**Comment by der** [comment]
Thanks but trying to run that build gives:
```
java.lang.ClassNotFoundException: net.logstash.logback.encoder.LogstashEncoder
```

**Comment by simonoakesepimorphics** [comment]
Not sure why the previous snapshot didn't work, I suspect I deployed from main without pulling the changes. This one should work: s3://epi-repository/snapshot/com/epimorphics/lda/elda-common/3.0.1-SNAPSHOT/elda-common-3.0.1-20260703.131522-2.war

**Comment by der** [comment]
Confirm that this works, though setting the logback configuration is painful.

Problem with missing LogstashEncoder was due to me trying to set logback config location using a system property which was being picked up by tomcat instead of the war.

Trying installing a config into the classpath resources root in the war (i.e. `WEB-INF/classes`) but is wasn't picked up.

Seems the elda-common war has a `logback.xml` in the root of the war itself which seems odd but was taking precedence so had to rewrite the war to replace that file to get it to take.

At that point request id logging is working in local test.

**Comment by kal** [comment]
@der - I think I added the logback.xml to the root of the project so that logging is configured by default. I didn't realise the precedence rule would make it difficult to override in deployment (that in itself seems odd to me). If it is better to have it omitted and be part of any deployment configuration we could open a separate ticket for that.

**Comment by der** [comment]
Makes sense to have a default one baked in but worth documenting where it is to avoid people being caught out.
